src/examples/hotgym/HelloSPTP.cpp

```diff
@@ -148,19 +155,22 @@ EPOCHS = 2; // make test faster in Debug
     //Anomaly (pure x likelihood)
     an = tm.anomaly;
     avgAnom10.compute(an); //moving average
-    if(e % 1000 == 0) {
+    if(e % (500 + avgAnom10.getData().size() /*size of avg window for an*/) == 0) {
       NTA_CHECK(avgAnomOld_ >= avgAnom10.getCurrentAvg()) << "TM should learn and avg anomalies improve, but we got: "
         << avgAnomOld_ << " and now: " << avgAnom10.getCurrentAvg(); //invariant
       avgAnomOld_ = avgAnom10.getCurrentAvg(); //update
     }
-    tAnLikelihood.start();
-    anLikelihood.anomalyProbability(an); //FIXME AnLikelihood is 0.0, probably not working correctly
-    tAnLikelihood.stop();
+
+    //Classifier, Predictor
+    tCls.start();
+    //! pred.learn(e, outTM, { realToCategory_(data) }); //FIXME fails with bad_alloc in Release, no crash in Debug?!
```

Comment:
a strange crash in Predictor. Runs fine in Debug mode, but crashes with std::bad_alloc in Release (I cant get much debug info from Release). @dkeeney any idea how to debug? Can you replicate that behavior? 
@ctrl-z-9000-times am I using the Predictor correcly? Thoughts where it crashes?